packages/directlinespeech/src/createAdapters.js

```diff
@@ -155,6 +166,31 @@ export default async function create({
     }, TOKEN_RENEWAL_INTERVAL);
   }
 
+  // Renew token per interval.
+  if (enableInternalHttpSupport) {
+    const interval = setInterval(async () => {
+      // #2660 If the connector has been disposed, we should stop renewing the token.
+
+      // TODO: We should use a public implementation if Speech SDK has one related to "privIsDisposed".
+      if (dialogServiceConnector.privIsDisposed) {
+        clearInterval(interval);
+      }
+
+      const refreshedDirectLineToken = await refreshDirectLineToken(directLineToken);
+
+      if (!refreshedDirectLineToken) {
+        return console.warn(
+          'botframework-directlinespeech-sdk: Renew token failed because call to refresh token Direct Line API did not return a new token.'
+        );
+      }
+
+      config.setProperty(PropertyId.Conversation_ApplicationId, refreshedDirectLineToken);
+      dialogServiceConnector = patchDialogServiceConnectorInline(new DialogServiceConnector(config, audioConfig));  // eslint-disable-line require-atomic-updates
```

Comment:
The original `dialogServiceConnector` object has already passed to `new DirectLineSpeech` constructor. Re-assigning to the `dialogServiceConnector` will not get it into the `DirectLineSpeech` instance.

Can you explain why a re-assignment is needed?

Reply:
my bad, only needed to update the config property and re-connect.